.github/autolabeler.yml

```diff
@@ -0,0 +1,115 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Bot page: https://github.com/apps/probot-autolabeler
+# The matching patterns follow the .gitignore spec.
+# See: https://git-scm.com/docs/gitignore#_pattern_format
+# Also, note that the plugin uses 'ignore' package. See also
+# https://github.com/kaelzhang/node-ignore
+INFRA:
+  - ".github/"
+  - "appveyor.yml"
+  - "/tools/"
+  - "/dev/create-release/"
+  - ".asf.yaml"
+  - ".gitattributes"
+  - ".gitignore"
+BUILD:
+  - "/dev/"
+  - "/build/"
+  - "/project/"
+  - "/assembly/"
+  - "*pom.xml"
+  - "/bin/docker-image-tool.sh"
+  - "/bin/find-spark-home*"
+  - "scalastyle-config.xml"
+DOCS:
+  - "docs/"
+  - "/README.md"
+  - "/CONTRIBUTING.md"
+EXAMPLES:
+  - "examples/"
+  - "/bin/run-example*"
+CORE:
+  - "/core/"
+  - "/common/kvstore"
+  - "/common/network-common"
+  - "/common/network-shuffle"
+  - "/python/pyspark/*.py"
+  - "/python/pyspark/tests/*.py"
+  - "/sbin/*master*.sh"
+  - "/sbin/*slave*.sh"
+  - "/sbin/spark-config.sh"
+  - "/sbin/*daemon*.sh"
+  - "/sbin/*history*.sh"
+  - "/sbin/*mesos*.sh"
+SPARK SUBMIT:
+  - "/bin/spark-submit*"
+SPARK SHELL:
+  - "/repl"
+SQL:
+  - "sql/"
+  - "/common/unsafe"
+  - "/python/pyspark/sql"
+  - "!/python/pyspark/sql/avro"
+  - "!/python/pyspark/sql/streaming.py"
+  - "!/python/pyspark/sql/tests/test_streaming.py"
+  - "/bin/spark-sql*"
+  - "/bin/beeline*"
+  - "/sbin/*thriftserver*.sh"
+AVRO:
+  - /external/avro"
+  - "/python/pyspark/sql/avro"
+DSTREAM:
+  - "/streaming/"
+  - "/data/streaming"
+  - "/external/flume/"
+  - "/external/flume-assembly/"
+  - "/external/flume-sink/"
+  - "/external/kinesis-asl/"
+  - "/external/kinesis-asl-assembly/"
+  - "/external/kafka-0-10/"
+  - "/external/kafka-0-10-assembly/"
+  - "/python/pyspark/streaming"
+GRAPHX:
+  - "/graphx/"
+  - "/data/graphx"
+ML:
+  - "ml/"
+  - "/python/pyspark/ml"
+MLLIB:
+  - "spark/mllib/"
+  - "/mllib-local"
+  - "/python/pyspark/mllib"
+STRUCTURED STREAMING:
+  - "sql/**/streaming/"
+  - "/external/kafka-0-10-sql/"
+  - "/python/pyspark/sql/streaming.py"
+  - "/python/pyspark/sql/tests/test_streaming.py"
+PYTHON:
+  - "/bin/pyspark*"
+  - "python/"
+R:
+  - "r/"
+  - "R/"
```

Comment:
According to https://github.com/kaelzhang/node-ignore#optionsignorecase-since-400 which the [plugin uses](https://github.com/mithro/autolabeler/blob/master/index.js#L22), seems it's case-insensitive by default but I just kept it.